pkg/installation.go

```diff
@@ -30,33 +26,20 @@ func CreateInstallationResources(cl client.Client, scheme *runtime.Scheme, log l
 	tektonInstallation := tektoninstallation.NewInstallation()
 	cheInstallation := cheinstallation.NewInstallation()
 
-	subscriptions := &v1alpha1.SubscriptionList{}
-	err := cl.List(context.TODO(), subscriptions, client.InNamespace(defaultOpenShiftOperatorsNamespace))
-	if err != nil {
-		return errors.Wrap(err, "unable to list the subscriptions")
-	}
-	for _, subscription := range subscriptions.Items {
-		if subscription.Spec.Package == codereadyToolchainPackageName {
-			if err := controllerutil.SetControllerReference(&subscription, tektonInstallation, scheme); err != nil {
-				return errors.Wrap(err, "unable to set the owner reference to Tekton Installation")
-			}
-			if err := controllerutil.SetControllerReference(&subscription, cheInstallation, scheme); err != nil {
-				return errors.Wrap(err, "unable to set the owner reference to Che Installation")
-			}
-			break
-		}
-	}
+	processor := template.NewProcessor(cl, scheme)
+
+	// we cannot set the owner reference for the *Installation resources because fo this issue: https://issues.redhat.com/browse/CRT-454
 
 	// create the TektonInstallation resource, stop if something wrong happened
 	log.Info("Creating the Tekton installation resource")
-	if err = cl.Create(context.TODO(), tektonInstallation); err != nil && !apierrors.IsAlreadyExists(err) {
+	if _, err := processor.ApplySingle(tektonInstallation, true, nil); err != nil {
```

Comment:
I'm sure it works, but the use of the template processor is a bit confusing, since the `terktonInstallation` is not a template per-se.

Reply:
well, I need to update the already existing CRs to make sure that the owner reference is removed when the operator is being updated. 
I can rename the processor in the common, or slightly refactor the functions there if you think that it would make more sense and would be less confusing

Reply:
yes, maybe it would make sense. Maybe just calling `CreateOrUpdate(..)` would clarify? (but this method needs to be package-exposed, first)

Reply:
WDYT? https://github.com/codeready-toolchain/toolchain-common/pull/74

Reply:
```
applyClient := applyCl.NewApplyClient(cl, scheme)
applyClient.CreateOrUpdateObject(.....)
```
Looks good to me ^

Reply:
yes, it looks much clearer! thanks for making this refactor effort @MatousJobanek ;)

Reply:
done in CreateOrUpdateObject